src/main/scala/devices/spi/SPIPhysical.scala

```diff
@@ -59,11 +112,22 @@ class SPIPhysical(c: SPIParamsBase) extends Module {
     Mux(fmt.endian === SPIEndian.MSB, data, Cat(data.toBools))
 
   val rxd = Cat(io.port.dq.reverse.map(_.i))
-  val samples = Seq(rxd(1), rxd(1, 0), rxd)
+  val rxd_delayed = Vec(Seq.fill(io.port.dq.size)(false.B))
+
+  //Adding fine-granularity delay buffers on the received data
+  val fine_grain_delay = Seq.fill(io.port.dq.size) {Module(new BlackBoxDelayBuffer())}
+  for (j <- 0 to (io.port.dq.size - 1)) { 
+    fine_grain_delay(j).io.in := rxd(j)
+    fine_grain_delay(j).io.sel := io.ctrl.extradel.fine
```

Comment:
what if these aren't the same width? I think chisel will tie the upper bits of `fine_grain_delay.io.sel` to 0, but it's probably not good to rely on that. What if user specifies they want more `fineDelayBits` than 6?